lib/json_matchers/matcher.rb

```diff
@@ -10,16 +10,22 @@ def initialize(schema_path, **options)
     def matches?(response)
       @response = response
 
-      JSON::Validator.validate!(
-        schema_path.to_s,
-        response.body,
-        options,
-      )
-    rescue JSON::Schema::ValidationError => ex
-      @validation_failure_message = ex.message
-      false
-    rescue JSON::ParserError
-      raise InvalidSchemaError
+      begin
+        add_schemata_to_document_store
```

Comment:
It feels weird to add the schemata to the document store when `matches?` is called. I'm sure there's a better place for it but I'm not sure where. Perhaps when the matcher is defined?